webapp/src/errorList.tsx

```diff
@@ -29,27 +29,37 @@ export class ErrorList extends React.Component<ErrorListProps, ErrorListState> {
     }
 
     render() {
+        const showAvailableErrors = this.state.errors.length != 0;
         const showCollapseButton = true;
         const collapseTooltip = lf("Collapse Error List");
         function errorKey(error: pxtc.KsDiagnostic): string {
             // React likes have a "key" for each element so that it can smartly only
             // re-render what changes. Think of it like a hashcode/
             return `${error.messageText}-${error.fileName}-${error.line}-${error.column}`
         }
-        return <div className="errorList" >
-            {showCollapseButton &&
+            
+        return <div className="errorList" > 
+            {showCollapseButton && showAvailableErrors &&
                 <sui.Button id='toggleErrorList' className={`toggleErrorList collapse-button large`}
-                    icon={`inverted chevron ${this.state.isCollapsed ? 'up' : 'down'}`}
-                    title={collapseTooltip} onClick={this.onCollapseClick} />}
-            <div className="errorListInner" hidden={this.state.isCollapsed}>
-                {
+                icon={`inverted chevron ${this.state.isCollapsed ? 'up' : 'down'}`}
+                title={collapseTooltip} onClick={this.onCollapseClick} />}
+            {showAvailableErrors
```

Comment:
You probably don't need to branch here. Instead, you could just add a `|| lf("You have no errors")` to the  end of the map.

Also if things seem to get too messy feel free to break stuff out into functions or methods. Regular functions can return JSX/react syntax in .tsx files.